File: src/app/shared/models/job.ts


import { Validators } from '@angular/forms';
import { Role } from './role';

export class Job {
  jobId: number;
  jobName: string;
  salary: string;
  jobDescription: string;
  vacancy: string;
  jobTypeControl: string;
  company: string;
  role: Role;

  static formValidationRules = {
    jobName: ['', [Validators.required, Validators.minLength(5)]],
    salary: ['', [Validators.required, Validators.minLength(4)]],
    jobDescription: ['', [Validators.required, Validators.minLength(8)]],
    vacancy: ['', [Validators.required, Validators.minLength(4)]],
    jobTypeControl: ['', [Validators.required]],
    company: ['', [Validators.required]],
    role: ['', [Validators.required]],
  };

  constructor(data?: any) {
    if (data) {
      this.jobId = data.jobId || '';
      this.jobName = data.jobName || '';
      this.salary = data.salary || '';
      this.jobDescription = data.jobDescription || '';
      this.vacancy = data.vacancy || '';
      this.jobTypeControl = data.jobTypeControl || '';
      this.company = data.company || '';
      this.role = data.role || null;
    } else {
      this.jobId = 0;
      this.jobName = '';
      this.salary = '';
      this.jobDescription = '';
      this.vacancy = '';
      this.jobTypeControl = '';
      this.company = '';
      this.role =null;
    }
  }
}

File: online_recruitment_back/src/main/java/com/infosys/entities/Job.java
package com.infosys.entities;

// import com.fasterxml.jackson.annotation.JsonIgnore;
import com.fasterxml.jackson.annotation.JsonIgnoreProperties;

import jakarta.persistence.Entity;
import jakarta.persistence.FetchType;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.GenerationType;
import jakarta.persistence.Id;
import jakarta.persistence.JoinColumn;
import jakarta.persistence.ManyToOne;
@JsonIgnoreProperties({"hibernateLazyInitializer","handler"})
@Entity
public class Job {
	
	@Id
	@GeneratedValue(strategy=GenerationType.AUTO)
	private int jobId;
	private String jobName;
	private String jobTypeControl;
	private String description;
	private int vacancy;
	private double salary;
	private String company;
	
	
	
	@ManyToOne(fetch = FetchType.EAGER)
    @JoinColumn(name = "roleId")
    private Role roles;
	
	private String location;
	
	public Job() {
		super();
	}

	public Job(int jobId, String jobName, String jobTypeControl, String description, int vacancy, double salary,
			String company, Role roles, String location) {
		super();
		this.jobId = jobId;
		this.jobName = jobName;
		this.jobTypeControl = jobTypeControl;
		this.description = description;
		this.vacancy = vacancy;
		this.salary = salary;
		this.company = company;
		this.roles = roles;
		this.location = location;
	}

	public int getJobId() {
		return jobId;
	}

	public void setJobId(int jobId) {
		this.jobId = jobId;
	}

	public String getJobName() {
		return jobName;
	}

	public void setJobName(String jobName) {
		this.jobName = jobName;
	}

	public String getJobTypeControl() {
		return jobTypeControl;
	}

	public void setJobTypeControl(String jobTypeControl) {
		this.jobTypeControl = jobTypeControl;
	}

	public String getDescription() {
		return description;
	}

	public void setDescription(String description) {
		this.description = description;
	}

	public int getVacancy() {
		return vacancy;
	}

	public void setVacancy(int vacancy) {
		this.vacancy = vacancy;
	}

	public double getSalary() {
		return salary;
	}

	public void setSalary(double salary) {
		this.salary = salary;
	}

	public String getCompany() {
		return company;
	}

	public void setCompany(String company) {
		this.company = company;
	}

	public Role getRoles() {
		return roles;
	}

	public void setRoles(Role roles) {
		this.roles = roles;
	}

	public String getLocation() {
		return location;
	}

	public void setLocation(String location) {
		this.location = location;
	}

	@Override
	public String toString() {
		return "Job [jobId=" + jobId + ", jobName=" + jobName + ", jobTypeControl=" + jobTypeControl + ", description="
				+ description + ", vacancy=" + vacancy + ", salary=" + salary + ", company=" + company + ", roles="
				+ roles + ", location=" + location + "]";
	}

	
}
